src/components/Visualizer.svelte

```diff
@@ -52,13 +53,15 @@
         shaders.frag.last = shaders.frag.text;
         shaders.vert.last = shaders.vert.text;
       }
+      renderer.startRender();
       renderer.compile(shaders.frag.last, shaders.vert.last, models[selected]);
+      if (report) sendData('USER_ID', await renderer.getImage(), shaders.frag.last);
+      status = 'success';
     } catch (e) {
+      renderer.clear();
+      if (report) sendData('USER_ID', new Blob([]), shaders.frag.last);
```

Comment:
*Really* unsatisfied with this, but I'm not sure if there's a better way that isn't a lot of work.

Currently, the client sends the gator code to the server twice - once to compile, and another time to report current status.

This happens because before we compile we don't actually have all the data we need to report yet - as we need to compile in order to have an updated image available. So we first have to send the gator code to compile it, then actually have an image, and then we send the whole gator code again to report current state. Not sure if there's a simple way to improve this, but I might be missing something super obvious.

Reply:
Not super worried about this, though, because the weight of the gator code pales in comparison to the weight of the image.